### DISCUSSION TITLE
Host is missing for all OAuth providers redirect URL

### DISCUSSION THREAD

**Discussion by PickRelated** [strapi/strapi#6500] [Legacy Topics]
**Describe the bug**
Host is missing in redirect URL for all providers. As a result - app fails to auth with OAuth providers because of malformed URL.

With discord as an example:
Actual URL strapi redirects to: `https://discordapp.com/api/oauth2/authorize?client_id=<MY_ID>&response_type=code&redirect_uri=%2Fconnect%2Fdiscord%2Fcallback&scope=identify%20email`
Answer:
```json
{"redirect_uri": ["Not a well formed URL."]}
```

**Steps to reproduce the behavior**

1. Go to `/admin/plugins/users-permissions/providers`
2. Click on any provider
3. Scroll down to `The redirect URL to add in your <whatever> application configurations`
4. See server host missing

**Expected behavior**
Redirect URL should display host URL. Also strapi app should be able to authenticate user with OAuth providers.

**Screenshots**
![image](https://user-images.githubusercontent.com/12638230/83465313-3593cc80-a47c-11ea-9eaf-710af2a14d4f.png)

**Code snippets**
I have tried adding this to `.env` file.
```yaml
HOST=0.0.0.0
PORT=1337
URL=myhost.com
BACKEND_URL=myhost.com
FRONTEND_URL=myhost.com
```

But `strapi.backendURL` is undefined. I believe it is used to form callback URL for OAuth providers [here](https://github.com/strapi/strapi/blob/ba6c0b4ba2a1acbd6979456d394eb7f602494baa/packages/strapi-plugin-users-permissions/admin/src/components/PopUpForm/index.js#L55)

**System**

- Node.js version: 12.16.3
- NPM version: 6.9.0
- Strapi version: 3.0.1
- Database: sqlite
- Operating system: Mac OS 10.14.6

**Additional context**
I did not find anything about specifying backend url in documentation.

P.S. Thanks for strapi. It's cool! Cheers.

**Comment by derrickmehaffy** [comment]
https://strapi.io/documentation/v3.x/concepts/configurations.html#server

Those domains are extracted from the actual proxy settings, you won't be able to set them directly.

I'm moving this to a discussion as it's not a bug.

**Reply by PickRelated** [reply]
Thanks for your help.

I found out you need to set `url` experimentally. It was not obvious to me at all that this parameter is required for OAuth to work. Would be cool to see it in the docs.

```javascript
module.exports = ({ env }) => ({
  host: env('HOST', '0.0.0.0'),
  url: 'http://mysite.com',
  port: env.int('PORT', 1337),
});
```

But callback URL on the provider popup is still `/connect/discord/fallback`. Since this popup advises to paste this into OAuth provider's trusted redirect URL - people will get confused. So is this a bug after all?

![image](https://user-images.githubusercontent.com/12638230/83540537-29058780-a501-11ea-868f-edd2bca21952.png)

**Reply by techguydave** [reply]
It's definitely a bug, it's covered by #6570.
